packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWTextInput.tsx

```diff
@@ -0,0 +1,257 @@
+import React, { FC } from 'react';
+import { CheckCircle, Warning } from '@phosphor-icons/react';
+
+import { CWLabel } from '../cw_label';
+import { CWText } from '../cw_text';
+import type { ValidationStatus } from '../cw_validation_text';
+import { getClasses } from '../helpers';
+import { ComponentType } from '../types';
+
+import 'components/component_kit/new_designs/CWTextInput.scss';
+
+type TextInputSize = 'small' | 'large';
+
+export type BaseTextInputProps = {
+  autoComplete?: string;
+  autoFocus?: boolean;
+  containerClassName?: string;
+  defaultValue?: string | number;
+  value?: string | number;
+  iconLeft?: JSX.Element;
+  iconLeftonClick?: () => void;
+  iconRight?: JSX.Element;
+  iconRightonClick?: () => void;
+  inputValidationFn?: (value: string) => [ValidationStatus, string] | [];
+  label?: string | React.ReactNode;
+  maxLength?: number;
+  isCompact?: boolean;
+  name?: string;
+  onInput?: (e) => void;
+  onenterkey?: (e) => void;
+  onClick?: (e) => void;
+  placeholder?: string;
+  tabIndex?: number;
+  instructionalMessage?: string;
+  manualStatusMessage?: string;
+};
+
+type InputStyleProps = {
+  inputClassName?: string;
+  darkMode?: boolean;
+  disabled?: boolean;
+  size?: TextInputSize;
+  fullWidth?: boolean;
+  validationStatus?: ValidationStatus;
+  displayOnly?: boolean;
+};
+
+type InputInternalStyleProps = {
+  hasLeftIcon?: boolean;
+  hasRightIcon?: boolean;
+};
+
+type TextInputProps = BaseTextInputProps &
+  InputStyleProps &
+  InputInternalStyleProps &
+  React.HTMLAttributes<HTMLDivElement>;
+
+type MessageRowProps = {
+  hasFeedback?: boolean;
+  label?: string | React.ReactNode;
+  statusMessage?: string;
+  validationStatus?: ValidationStatus;
+  instructionalMessage?: string;
+};
+
+export const MessageRow: FC<MessageRowProps> = ({
```

Comment:
break this file into multiple files so the structure look like this

components/component_kit/new_designs/CWTextInput => directory, and inside:
- CWTextInput.tsx => file component
- CWTextInput.scss => style file 
- MessageRow.tsx => file component
- MessageRow.scss => style file
- useTextInputWithValidation.ts => hook file
- index.ts => import/export of CWTextInput

Reply:
Addressed ([here](https://github.com/hicommonwealth/commonwealth/commit/9da3b3daf835b25e7f1c3fb22d33b6d8ef916a2e) and [here](https://github.com/hicommonwealth/commonwealth/commit/0a37f530267f738895420680922b83f035da3bb9)) and tested on
- Storybook
- /components page
- edit profile page